### ISSUE TITLE
Consider removing zbar dependency

### ISSUE THREAD

**Issue by ZeppLu** [shadowsocks/shadowsocks-qt5#430]
zbar 已经好几年没更新过了，而且依赖的是 qt4，为了一个 zbar 装 qt4 有点不值。
不妨考虑用 [qrencode](https://github.com/fukuchi/libqrencode)，但是不能 decode 二维码。
其实从屏幕读取二维码的操作我几乎从没用过，就算需要，也可以用浏览器的插件解码。
或者也可以 fork zbar，升级到 qt5.

**Comment by librehat** [comment]
`qrencode` has been used already.
I tend to replace `zbar` with a bundled [quirc](https://github.com/dlbeer/quirc) since it's not shipped by most Linux distros. :(

**Comment by ZeppLu** [comment]
供参考：KDE 那帮家伙搞出了个新玩意 [prison](https://api.kde.org/frameworks-api/frameworks-apidocs/frameworks/prison/html/)。
从[这里](https://www.archlinux.org/packages/extra/x86_64/prison/)看来运行时不依赖 KDE 框架（编译需要 extra-cmake-module），基于 libdmtx 与 qrencode，不太清楚 windows 上是否可用

**Comment by librehat** [comment]
你看一下介绍，这个是用来生成二维码的。zbar是用来解码的！
生成二维码我们直接使用了qrencode

**Comment by ZeppLu** [comment]
抱歉，，没仔细看

**Comment by baimafeima** [comment]
@librehat Any updates regarding the zbar issue?

**Comment by ZeppLu** [comment]
@meoow 关键是 Linux 下包之间的依赖问题，ss-qt5 依赖 zbar，zbar 又依赖 qt4. 如果自己编译打包安装当然可以避免这个问题，但更新不方便，而且能用包管理器的地方还是用来得舒心。

**Comment by baimafeima** [comment]
For the latest 3.0.0 version, will zbar be replaced with an alternative?

**Comment by librehat** [comment]
@baimafeima Sorry I didn't have time replacing `zbar` with something else

**Comment by librehat** [comment]
Some candidates for replacing zbar:
1. https://github.com/glassechidna/zxing-cpp
2. https://github.com/dlbeer/quirc

I'm more inclined to replace `zbar` with `zxing-cpp` since `zxing` has been widely adopted and tested, it is also a CMake based project which means it is very easy to be packaged. However, they haven't released any stable versions, which might be a problem for a lot of distributions to ship it in their official repository.

**Comment by baimafeima** [comment]
@librehat It looks like `zxing-cpp` is not maintained anymore.

**Comment by librehat** [comment]
Without a good candidate, I'm afraid we can't just remove zbar yet. I'd suggest you package zbar without Qt4

**Comment by tenten8401** [comment]
@baimafeima The link that @librehat posted is still being maintained, although the C++ port is not an "official" project anymore.

**Comment by baimafeima** [comment]
@librehat Any updates on the zbar issue? What about getting a flatpak out in the meantime? It's recently reached 1.0, so you may want to take another look: https://flatpak.org/press/2018-08-20-flatpak-1.0/

**Comment by librehat** [comment]
@baimafeima Can't you use AppImage? Not all distributions support `flatpak` out of box

**Comment by baimafeima** [comment]
@librehat Yes, I could but I prefer not to use any executables but a sandboxed solution. As I am running Solus, flatpak and snaps are supported out of the box.
